### PR TITLE
Added Decimal to BCD and vice versa #2717

### LINKED ISSUE #2717: Decimal to BCD and vice versa #2
#2 

Currently, there is no decimal to BCD and vice versa conversion in CV.

**Describe the solution you'd like**
Screenshot 2021-12-15 at 8 45 05 PM

**Additional context**
examples:
```
16(10) = 10110 (BCD)
999 (10) = 100110011001 (BCD)

```

**Are you working on this? (Yes/No)**
Yes

### PR DESCRIPTION
**Pull request by pulkit-30**
Fixes #2717

#### Describe the changes you have made in this PR -
Added Decimal to Binary-coded decimal conversion with dec-bin-hex converter and vice-versa.

### Screenshots of the changes (If any) -
Screenshot 2021-12-15 at 8 45 05 PM

Please provide me with appropriate feedback so that I can continue working on this.

### CODE REVIEW

**Comment by pulkit-30** [comment]
Hey @tachyons,
Please analyze these changes and provide me with appropriate feedback so that I can continue working on this.

**Comment by pulkit-30** [comment]
Hey @tachyons  @gr455, Please review this PR and provide me with appropriate feedback so that I can continue working on this.

**Review by gr455 — CHANGES_REQUESTED** [review]
Works good, just fix codeclimate

**Comment by pulkit-30** [comment]
> Works good, just fix codeclimate

Hey @gr455, tried to fix codeclimate problem.
Now it shows Complexity Issue `File utils.js has 253 lines of code (exceeds 250 allowed). Consider refactoring. `
Screenshot 2021-12-25 at 2 17 18 PM

What can we do to solve this?

**Comment by gr455** [comment]
> > Works good, just fix codeclimate
> 
> Hey @gr455, tried to fix codeclimate problem.
> Now it shows Complexity Issue `File utils.js has 253 lines of code (exceeds 250 allowed). Consider refactoring. `
> Screenshot 2021-12-25 at 2 17 18 PM
> 
> What can we do to solve this?

I guess that one is fine. Let it be

File: simulator/src/utils.js [hunk 1]
```diff
@@ -268,6 +270,24 @@ export function setupBitConvertor() {
             x = parseInt(inp, 2);
         setBaseValues(x);
     })
+    $("#bcdInput").on('keyup', function () {
+        var inp = $("#bcdInput").val();
```
Comment (tachyons) [inline simulator/src/utils.js:274]:
```suggestion
        var input = $("#bcdInput").val();
```

File: simulator/src/utils.js [hunk 2]
```diff
@@ -268,6 +270,24 @@ export function setupBitConvertor() {
...
+        var x = 0;
```
Comment (tachyons) [inline simulator/src/utils.js:275]: Avoid single character variables

Comment (pulkit-30) [reply simulator/src/utils.js]: Thanks for the suggestion sir,
Replaced variable name `x` with `num`.

**Comment by tachyons** [comment]
Thanks 👍

**Pull request merged**